src/features/metadata-based-view/MetadataBasedItemList.js

```diff
@@ -95,23 +98,44 @@ class MetadataBasedItemList extends React.Component<Props, State> {
         };
     }
 
-    handleItemClick(item: BoxItem): void {
-        const { onItemClick }: Props = this.props;
+    getInitialState = () => {
+        return {
+            // initial MultiGrid load
+            editedColumnIndex: -1,
+            editedRowIndex: -1,
+            hoveredRowIndex: -1,
+            hoveredColumnIndex: -1,
+            valueBeingEdited: null,
+        };
+    };
+
+    getItemWithPermissions = (item: BoxItem): BoxItem => {
         /*
             - @TODO: Remove permissions object once its part of API response.
-            - In Content Explorer element, if can_preview permission is false, there is no action taken onClick(item).
-            - Until the response has permissions, add "can_preview: true" so that users can click to launch the Preview modal. If users don't have access, they will see the error when Preview loads.
+            - add "can_preview: true" so that users can click to launch the Preview modal. If users don't have access, they will see the error when Preview loads.
+            - add "can_upload: true" so that users can update the metadata values.
         */
-        const permissions = { can_preview: true };
-        const itemWithPreviewPermission = { ...item, permissions };
+        const permissions = {
+            can_preview: true,
+            can_upload: true,
+        };
+        return { ...item, permissions };
+    };
 
-        onItemClick(itemWithPreviewPermission);
+    getMetadataColumnName(column: MetadataColumnConfig | string): string {
```

Comment:
Can we add tests for `getMetadataColumnName`, `handleEditIconClick`, and `handleSave`?

Reply:
Done